src/java/org/apache/cassandra/tools/NodeTool.java

```diff
@@ -316,6 +142,58 @@ private static void printHistory(String... args)
         }
     }
 
+
+    public static List<String> getCommandsWithoutRoot(String separator)
+    {
+        List<String> commands = new ArrayList<>();
+        getCommandsWithoutRoot(createCommandLine(new CassandraCliFactory(new NodeProbeFactory(), Output.CONSOLE)), commands, separator);
+        return commands;
+    }
+
+    private static void getCommandsWithoutRoot(CommandLine cli, List<String> commands, String separator)
+    {
+        String name = cli.getCommandSpec().qualifiedName(separator);
+        // Skip the root command as it's not a real command.
+        if (cli.getCommandSpec().root() != cli.getCommandSpec())
+            commands.add(name.replace(cli.getCommandSpec().root().qualifiedName() + separator, ""));
+        for (CommandLine sub : cli.getSubcommands().values())
+            getCommandsWithoutRoot(sub, commands, separator);
+    }
+
+    private static CommandLine createCommandLine(CassandraCliFactory factory)
+    {
+        return new CommandLine(new TopLevelCommand(), factory)
+                   .addMixin(JmxConnect.MIXIN_KEY, factory.create(JmxConnect.class))
+                   .setOut(new PrintWriter(factory.output.out, true))
+                   .setErr(new PrintWriter(factory.output.err, true));
+    }
+
+    private static void configureCliLayout(CommandLine commandLine)
+    {
+        CliLayout defaultLayout = CliLayout.valueOf(toUpperCaseLocalized(CASSANDRA_CLI_LAYOUT.getDefaultValue()));
+        CliLayout layoutEnv = CassandraRelevantEnv.CASSANDRA_CLI_LAYOUT.getEnum(true, CliLayout.class,
+                                                                                CASSANDRA_CLI_LAYOUT.getDefaultValue());
+        CliLayout layoutSysEnv = CASSANDRA_CLI_LAYOUT.getEnum(true, CliLayout.class);
+        CliLayout layout;
+        if (layoutEnv == layoutSysEnv)
+            layout = layoutEnv;
+        else
+            layout = layoutEnv == defaultLayout ? layoutSysEnv : layoutEnv;
+        switch (layout)
+        {
+            case AIRLINE:
+                commandLine.setHelpFactory(CassandraCliHelpLayout::new)
+                           .setUsageHelpWidth(CassandraCliHelpLayout.DEFAULT_USAGE_HELP_WIDTH)
+                           .setHelpSectionKeys(CassandraCliHelpLayout.cassandraHelpSectionKeys());
+                break;
+            case PICOCLI:
+                break;
+            default:
+                throw new IllegalStateException("Unknown CLI layout: " +
+                                                CASSANDRA_CLI_LAYOUT.getString());
```

Comment:
this should be `layout`  instead of `CASSANDRA_CLI_LAYOUT.getString()`

Reply:
Fixed.